### PR TITLE
Dtsw 4343 create dts devel code

### PR DESCRIPTION
**Pull request by Tuxliri**

### CODE REVIEW

File: devel/code/command.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+DEFAULT_REMOTE_SYNC_LOCATION = "/code"
+
+DEFAULT_TRUE = object()
+
+DEVCONTAINER_TEMPLATE = {
+    "name": "Duckietown Dev Container",
+    "image": "${{localEnv:DOCKER_REGISTRY}}/duckietown/${{localWorkspaceFolderBasename}}:ente-amd64",  # Replace with the actual Docker image tag
+    "workspaceFolder": f"{DEFAULT_REMOTE_SYNC_LOCATION}/src/${{localWorkspaceFolderBasename}}",
+    "mounts": [
+        *[f"source={m},target={m},type=bind" for m in DEFAULT_MOUNTS],
```
Comment (afdaniele) [inline devel/code/command.py]: Replace `[*[...]]` with `[...]`

File: devel/code/command.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        output_file_path = ".devcontainer/devcontainer.json"
+
+        # Create .devcontainer folder
+        if not os.path.exists('.devcontainer') and not os.path.isdir('.devcontainer'):
+            os.mkdir('.devcontainer')
+        with open(output_file_path, 'w') as json_file:
+            json.dump(DEVCONTAINER_TEMPLATE, json_file, indent=4,)
+
+        dtslogger.info(f'DEVCONTAINER_TEMPLATE written to {output_file_path}')
+        dtslogger.info("You can now open the devcontainer in your favorite editor")
```
Comment (afdaniele) [inline devel/code/command.py]: We can attempt to open VSCode in the project's location.

File: devel/code/command.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
+import argparse
```
Comment (afdaniele) [inline devel/code/command.py]: More than `dts devel code`, I would add this as a subcommand of the existing `dts vscode` as `dts vscode open`.

**Review by afdaniele — CHANGES_REQUESTED** [review]

File: devel/code/command.py (hunk 2 above)
Comment (Tuxliri) [reply devel/code/command.py]: Yes, there seems to be a shell api for devcontainers (other than the one already existing from VSCode) but wanted to keep this compatible with PyCharm as well. But we could look into opening VSCode already configured for easy development.

File: devel/code/command.py (hunk 3 above)
Comment (Tuxliri) [reply devel/code/command.py]: That's an option, I intending this exclusively for a certain type of development workflow for now but we can expand it.

File: devel/code/command.py (hunk 1 above)
Comment (Tuxliri) [reply devel/code/command.py:28]: It created a nested list without the `*`, then the json would not be according to spec.

File: devel/code/command.py (hunk 2 above)
Comment (Tuxliri) [reply devel/code/command.py]: Added.

**Review by Tuxliri — COMMENTED** [review]
Implemented requested changes:

- added automatic opening of vscode in container
- renamed command to `dts vscode open`

**Pull request merged**